runtime/reference/ts_config_migration.md

```diff
@@ -0,0 +1,501 @@
+---
+title: "Configuring TypeScript"
+oldUrl:
+  - /runtime/manual/advanced/typescript/faqs/
+  - /runtime/manual/advanced/typescript/migration/
+  - /runtime/manual/advanced/typescript/configuration/
+  - /runtime/manual/advanced/typescript/types/
+  - /runtime/manual/typescript/types/
+  - /runtime/manual/advanced/faqs/
+---
+
+Deno’s flexibility shines in its equal treatment of TypeScript and JavaScript.
+Whether you’re transitioning from JavaScript to TypeScript or vice versa, Deno
+has features to ease the journey.
+
+## Type Checking JavaScript
+
+You may wish to make your JavaScript more type-sound without adding type
+annotations everywhere. Deno supports using the TypeScript type checker to type
+check JavaScript. You can mark individual files by adding the check JavaScript
+pragma to the file:
+
+```js
+// @ts-check
+```
+
+This will cause the type checker to infer type information about the JavaScript
+code and raise any issues as diagnostic issues.
+
+These can be turned on for all JavaScript files in a program by providing a
+configuration file with the check JS option enabled:
+
+```json
+{
+  "compilerOptions": {
+    "checkJs": true
+  }
+}
+```
+
+And then setting the `--config` option on the command line.
```

Comment:
Nit: This reads a bit odd as a sentence alone. It makes sense if it would be part of the previous one.